crates/ruff_linter/src/rules/flake8_pyi/rules/iter_method_return_iterable.rs

```diff
@@ -15,24 +15,46 @@ use crate::checkers::ast::Checker;
 /// `__iter__` methods should always should return an `Iterator` of some kind,
 /// not an `Iterable`.
 ///
-/// In Python, an `Iterator` is an object that has a `__next__` method, which
-/// provides a consistent interface for sequentially processing elements from
-/// a sequence or other iterable object. Meanwhile, an `Iterable` is an object
-/// with an `__iter__` method, which itself returns an `Iterator`.
+/// In Python, an `Iterable` is an object that has an `__iter__` method; an
+/// `Iterator` is an object that has `__iter__` and `__next__` methods. All
+/// `__iter__` methods are expected to return `Iterator`s in Python. Type
```

Comment:
I would probably remove the second "in Python", assuming that's clear from a python linter.
```suggestion
/// `__iter__` methods are expected to return `Iterator`s. Type
```